WTG.Analyzers/Analyzers/PublicTupleType/PublicTupleTypeAnalyzer.cs

```diff
@@ -0,0 +1,303 @@
+			context.RegisterSyntaxNodeAction(
+				c => Analyze(c, cache),
+				SyntaxKind.ConstructorDeclaration,
+				SyntaxKind.MethodDeclaration,
+				SyntaxKind.PropertyDeclaration,
+				SyntaxKind.IndexerDeclaration,
+				SyntaxKind.DelegateDeclaration,
+				SyntaxKind.EventDeclaration,
+				SyntaxKind.EventFieldDeclaration,
+				SyntaxKind.FieldDeclaration);
+		}
+
+		static void Analyze(SyntaxNodeAnalysisContext context, FileDetailCache cache)
+		{
+			if (cache.IsGenerated(context.SemanticModel.SyntaxTree, context.CancellationToken))
+			{
+				return;
+			}
+
+			var decl = (MemberDeclarationSyntax)context.Node;
+			var processor = new Processor(context.SemanticModel, context.ReportDiagnostic, context.CancellationToken);
+			decl.Accept(processor);
+		}
+
+		sealed class Processor : CSharpSyntaxVisitor
+		{
+			public Processor(SemanticModel model, Action<Diagnostic> report, CancellationToken cancellationToken)
+			{
+				this.model = model;
+				this.report = report;
+				this.cancellationToken = cancellationToken;
+			}
+
+			public override void VisitConstructorDeclaration(ConstructorDeclarationSyntax node)
+			{
+				var syntax = model.GetDeclaredSymbol(node, cancellationToken);
+
+				if (syntax != null && syntax.IsExternallyVisible())
+				{
+					VisitParameters(node.ParameterList.Parameters);
+				}
+			}
+
+			public override void VisitMethodDeclaration(MethodDeclarationSyntax node)
+			{
+				if (!IsOverride(node.Modifiers))
+				{
+					var syntax = model.GetDeclaredSymbol(node, cancellationToken);
+
+					if (syntax != null && syntax.IsExternallyVisible() && !syntax.ImplementsAnInterface())
+					{
+						VisitParameters(node.ParameterList.Parameters);
+						Visit(node.ReturnType);
+					}
+				}
+			}
+
+			public override void VisitPropertyDeclaration(PropertyDeclarationSyntax node)
+			{
+				if (!IsOverride(node.Modifiers))
+				{
+					var syntax = model.GetDeclaredSymbol(node, cancellationToken);
+
+					if (syntax != null && syntax.IsExternallyVisible() && !syntax.ImplementsAnInterface())
+					{
+						Visit(node.Type);
+					}
+				}
+			}
+
+			public override void VisitFieldDeclaration(FieldDeclarationSyntax node)
+			{
+				var variables = node.Declaration.Variables;
+
+				if (variables.Count > 0)
+				{
+					var syntax = model.GetDeclaredSymbol(variables[0], cancellationToken);
+
+					if (syntax != null && syntax.IsExternallyVisible())
+					{
+						Visit(node.Declaration.Type);
+					}
+				}
+			}
+
+			public override void VisitEventDeclaration(EventDeclarationSyntax node)
+			{
+				if (!IsOverride(node.Modifiers))
+				{
+					var syntax = model.GetDeclaredSymbol(node, cancellationToken);
+
+					if (syntax != null && syntax.IsExternallyVisible() && !syntax.ImplementsAnInterface())
+					{
+						Visit(node.Type);
+					}
+				}
+			}
+
+			public override void VisitEventFieldDeclaration(EventFieldDeclarationSyntax node)
+			{
+				if (!IsOverride(node.Modifiers))
+				{
+					if (MustComply(node.Declaration))
```

Comment:
This could be collapsed into a single if block

Reply:
I was mostly just keeping it consistent with the others :)

Reply:
I would be up for changing the others :)